workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QualifiedLeadsTemplateValidatorController.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.partner;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import java.net.URL;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * @author Eduardo Diniz
+ */
+@RequestMapping("/file-validation")
+@RestController
+public class QualifiedLeadsTemplateValidatorController
+	extends BaseRestController {
+
+	@PostMapping("/validate")
+	public ResponseEntity<String> validateFile(
+		@AuthenticationPrincipal Jwt jwt,
+		@RequestParam("file") MultipartFile file,
+		HttpServletRequest httpServletRequest) {
+
+		try {
+			Set<String> fileColumns = _extractColumns(file);
+			Set<String> templateColumns = _extractColumnsFromTemplate();
+
+			if (fileColumns.equals(templateColumns)) {
+				return ResponseEntity.ok("File is valid.");
+			}
+
+			return ResponseEntity.badRequest(
+			).body(
+				"File is invalid. Columns do not match the template."
+			);
+		}
+		catch (IOException ioException) {
+			return ResponseEntity.status(
+				500
+			).body(
+				"Error processing file: " + ioException.getMessage()
+			);
+		}
+	}
+
+	private Set<String> _extractColumns(MultipartFile file) throws IOException {
+		String contentType = file.getContentType();
+
+		if ((contentType != null) && contentType.equals("text/csv")) {
+			return _extractColumnsFromCSV(file.getInputStream());
+		}
+
+		return _extractColumnsFromExcel(file.getInputStream());
+	}
+
+	private Set<String> _extractColumnsFromCSV(InputStream inputStream)
+		throws IOException {
+
+		Set<String> columns = new LinkedHashSet<>();
+
+		try (BufferedReader reader = new BufferedReader(
+				new InputStreamReader(inputStream))) {
+
+			String line;
```

Comment:
`String` objects should be initialized to either `null` or `""` blank string. in this case `null` would work